library/core/src/option.rs

```diff
@@ -1554,8 +1554,11 @@ impl<T> Option<T> {
     pub fn insert(&mut self, value: T) -> &mut T {
         *self = Some(value);
 
-        // SAFETY: the code above just filled the option
-        unsafe { self.as_mut().unwrap_unchecked() }
```

Comment:
Was there some problem you were hitting because of the function calls?  Is there a test you could add to detect whatever it was, to keep it from happening again in future?

Especially with MIR inlining on for `core` now, it's not clear to me that open-coding these is better than leaving them alone.

@rustbot author

Reply:
I wasn't having any specific problems, i just thought it might be better to merge the two functions into a single match statement, but if you think it would be better to just leave them, i can get rid of those changes.

Reply:
Without any specific reason, I'd like to just leave it as-is.  Otherwise it's equally valid to "might be better to use the clearer combinators instead of writing them out by hand", and could flip-flop back and forth in the implementation.

If you cut this down to just the forget-replace cleanup, then I'd approve it.

@rustbot author

Reply:
fair enough, i have reverted the other changes.